pkg/cluster/sync.go

```diff
@@ -340,7 +340,7 @@ func (c *Cluster) syncPatroniService() error {
 }
 
 func (c *Cluster) syncServices() error {
-	for _, role := range []PostgresRole{Master, Replica} {
+	for _, role := range []PostgresRole{c.masterRole(), c.replicaRole()} {
```

Comment:
another thing we need to think about is the transition period: e.g., when "master" is changed to something custom.
this way, we probably need to delete the old objects on sync and also remove those from the Cluster struct

otherwise
1. master -> meow change
2. sync event
3. creates service for the "meow" role (now there are both master and meow services)
4. c.Services[role] = createdSvc (where role == "meow" but we also already have c.Services["master"])

Reply:
and of course test the transition period at list manually:)